### PR TITLE
Saleor 2590 remove classes from cypress tests

### PR DESCRIPTION
**Pull request by karola312**
I want to merge this change because there are unnecessary classes in tests

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.
1. [ ] The changes are tested in different browsers (Chrome, Firefox, Safari).
1. [ ] The changes are tested in light and dark mode.

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2590 Remove unnecessary classes from cypress tests](https://app.clickup.com/t/2549495/SALEOR-2590)

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: cypress/apiRequests/Product.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import { getValueWithDefault } from "./utils/Utils";
 
-class Product {
-  getFirstProducts(first, search) {
-    const filter = search
-      ? `, filter:{
+export function getFirstProducts(first, search) {
```
Comment (krzysztofwolski) [inline cypress/apiRequests/Product.js:3]: When we are at refactoring - can we use gql tag like in queries in dashboard?

File: cypress/utils/collectionsUtils.js [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
-
-  deleteProperCollections(startsWith) {
-    cy.deleteProperElements(
-      this.collectionsRequest.deleteCollection,
-      this.collectionsRequest.getCollections,
-      startsWith,
-      "collection"
-    );
-  }
+export function deleteProperCollections(startsWith) {
```
Comment (krzysztofwolski) [inline cypress/utils/collectionsUtils.js:3]:
```suggestion
export function deleteCollectionsStartingWith(startsWith) {
```

Or, if we always create objects with `cypress` prefix, maybe helper which will be called `deleteTestCollections` would be better?

File: cypress/utils/productsUtils.js [hunk 3]
```diff
@@ -1,152 +1,130 @@
...
-import Product from "../apiRequests/Product";
+import * as attributeRequest from "../apiRequests/Attribute";
+import * as categoryRequest from "../apiRequests/Category";
+import * as productRequest from "../apiRequests/Product";
 
-class ProductsUtils {
-  productRequest = new Product();
-  attributeRequest = new Attribute();
-  categoryRequest = new Category();
+let product;
```
Comment (krzysztofwolski) [inline cypress/utils/productsUtils.js:5]: Having those objects on module leevel seems error-prone. Better create function which returns product or other object and save it in test scope.

File: cypress/integration/collections.js [hunk 4]
```diff
@@ -70,14 +64,13 @@ describe("Collections", () => {
     cy.visit(urlList.collections);
     let collection;
 
-    collectionsSteps
-      .createCollection(collectionName, false, defaultChannel)
+    createCollection(collectionName, false, defaultChannel)
```
Comment (krzysztofwolski) [inline cypress/integration/collections.js:67]: If every test need collection, you can create it in `beforeEach`

File: cypress/steps/channelsSteps.js [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
+) {
+  cy.get(CHANNELS_SELECTORS.createChannelButton)
+    .click()
+    .get(ADD_CHANNEL_FORM_SELECTORS.channelName)
+    .type(name)
+    .get(ADD_CHANNEL_FORM_SELECTORS.slug)
+    .type(slug)
+    .get(ADD_CHANNEL_FORM_SELECTORS.currency)
+    .click();
+  if (!otherCurrency) {
```
Comment (krzysztofwolski) [inline cypress/steps/channelsSteps.js:18]: Is other currency argumeent needed? This function could check if add currency option is available and click it if visible

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by jwm0 — APPROVED** [review]
Apart from what @krzysztofwolski mentioned, it looks good 👍

File: cypress/apiRequests/Product.js (hunk 1 above)
Comment (krzysztofwolski) [reply cypress/apiRequests/Product.js:3]: Investigated it a bit and more changes in making requests would be needed. It's a material for separate PR

File: cypress/utils/collectionsUtils.js (hunk 2 above)
Comment (karola312) [reply cypress/utils/collectionsUtils.js:3]: I rename this functions to deleteCollections/ProductsStartsWith. If in the future we would like to run tests in parallel, there shouldn't be always cypress prefix.

File: cypress/integration/collections.js (hunk 4 above)
Comment (karola312) [reply cypress/integration/collections.js:75]: These tests are creating collections with different options. First test is creating not published collection, second published, another one with different channel. I can move it to before each, and in test update created collection, but I don't think it's a good idea.

File: cypress/steps/channelsSteps.js (hunk 5 above)
Comment (karola312) [reply cypress/steps/channelsSteps.js]: It's not. I removed otherCurrency

File: cypress/integration/channels.js [hunk 6]
```diff
@@ -79,9 +76,9 @@ describe("Channels", () => {
 
   it("should validate slug name", () => {
```
[the remark this answers is not in the document]
Comment (karola312) [reply cypress/integration/channels.js:77]: I renamed it :)

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**